src/Nri/Ui/Button/V9.elm

```diff
@@ -0,0 +1,887 @@
+module Nri.Ui.Button.V9 exposing
+    ( button, link
+    , Attribute
+    , label, icon, custom
+    , onClick, href
+    , linkSpa, linkExternal, linkWithMethod, linkWithTracking, linkExternalWithTracking
+    , small, medium, large
+    , exactWidth, unboundedWidth, fillContainerWidth
+    , primary, secondary, danger, premium
+    , enabled, unfulfilled, disabled, error, loading, success
+    , delete
+    , toggleButton
+    )
+
+{-|
+
+
+# Changes from V8:
+
+  - Changes API to be attribute-based, rather than config-based
+
+
+# Create a button or link
+
+@docs button, link
+@docs Attribute
+@docs label, icon, custom
+
+
+## Add event listeners and hrefs
+
+@docs onClick, href
+@docs linkSpa, linkExternal, linkWithMethod, linkWithTracking, linkExternalWithTracking
+
+
+## Sizing
+
+@docs small, medium, large
+@docs exactWidth, unboundedWidth, fillContainerWidth
+
+
+## Change the color scheme
+
+@docs primary, secondary, danger, premium
+
+
+## Change the state (buttons only)
+
+@docs enabled, unfulfilled, disabled, error, loading, success
+
+
+# Commonly-used buttons
+
+@docs delete
+@docs toggleButton
+
+-}
+
+import Accessibility.Styled as Html exposing (Attribute, Html)
+import Accessibility.Styled.Role as Role
+import Accessibility.Styled.Widget as Widget
+import Css exposing (Style)
+import Css.Global
+import EventExtras.Styled as EventExtras
+import Html as RootHtml
+import Html.Styled as Styled
+import Html.Styled.Attributes as Attributes
+import Html.Styled.Events as Events
+import Json.Decode
+import Markdown.Block
+import Markdown.Inline
+import Nri.Ui
+import Nri.Ui.AssetPath as AssetPath exposing (Asset)
+import Nri.Ui.Colors.Extra as ColorsExtra
+import Nri.Ui.Colors.V1 as Colors
+import Nri.Ui.Fonts.V1
+import Nri.Ui.Html.Attributes.V2 as AttributesExtra
+import Nri.Ui.Svg.V1 as NriSvg exposing (Svg)
+import Svg
+import Svg.Attributes
+
+
+styledName : String -> String
+styledName suffix =
+    "Nri-Ui-Button-V9-" ++ suffix
+
+
+{-|
+
+    Button.button
+        [ Button.label "My great button!"
+        , Button.onClick ()
+        , Button.enabled
+        ]
+
+-}
+button : List (Attribute msg) -> Html msg
+button attributes =
+    List.foldl (\attribute b -> attribute b) build attributes
+        |> renderButton
+
+
+{-|
+
+    Button.link
+        [ Button.label "My great link!"
+        , Button.href "My href"
+        , Button.secondary
+        ]
+
+-}
+link : List (Attribute msg) -> Html msg
+link attributes =
+    List.foldl (\attribute l -> attribute l) build attributes
+        |> renderLink
+
+
+{-| -}
+label : String -> Attribute msg
+label label_ (ButtonOrLink config) =
+    ButtonOrLink { config | label = label_ }
+
+
+{-| -}
+icon : Svg -> Attribute msg
+icon icon_ (ButtonOrLink config) =
+    ButtonOrLink { config | icon = Just icon_ }
+
+
+{-| -}
+custom : Html.Attribute msg -> Attribute msg
```

Comment:
🎨 I wonder if `custom` should just take a list of `Html.Attribute msg`, just to make the calling code a little less redundant in the case that there are multiple custom attributes?

Reply:
Yeah I thought about that and waffled! I can do multiple 👍